File: autocare-backend/src/main/java/com/pobitra/autocare/service/VehicleService.java
package com.pobitra.autocare.service;

import com.pobitra.autocare.dto.VehicleRequestDTO;
import com.pobitra.autocare.dto.VehicleResponseDTO;

import java.util.List;

public interface VehicleService {

    // ==============================
    // CUSTOMER
    // ==============================

    // Add a vehicle for the logged-in customer
    VehicleResponseDTO createVehicle(
            VehicleRequestDTO dto,
            String email);

    // Get all vehicles of the logged-in customer
    List<VehicleResponseDTO> getMyVehicles(
            String email);

    // Update own vehicle
    VehicleResponseDTO updateVehicle(
            Long id,
            VehicleRequestDTO dto,
            String email);

    // Delete own vehicle
    void deleteVehicle(
            Long id,
            String email);

    // ==============================
    // ADMIN / STAFF
    // ==============================

    // View all vehicles
    List<VehicleResponseDTO> getAllVehicles();

    // View vehicle details
    VehicleResponseDTO getVehicleById(Long id);
}

File: autocare-frontend/src/services/vehicleService.js
import api from "./api";

const vehicleService = {

    // ===========================
    // CUSTOMER
    // ===========================

    // Get logged-in customer's vehicles
    getMyVehicles() {
        return api.get("/vehicles/my");
    },

    // Get one logged-in customer's vehicle
    getMyVehicleById(id) {
        return api.get(`/vehicles/my/${id}`);
    },

    // Add new vehicle
    createVehicle(vehicle) {
        return api.post("/vehicles", vehicle);
    },

    // Update own vehicle
    updateVehicle(id, vehicle) {
        return api.put(`/vehicles/${id}`, vehicle);
    },

    // Delete own vehicle
    deleteVehicle(id) {
        return api.delete(`/vehicles/${id}`);
    },

    // ===========================
    // ADMIN / STAFF
    // ===========================

    getAllVehicles() {
        return api.get("/vehicles");
    },

    getVehicleById(id) {
        return api.get(`/vehicles/${id}`);
    }

};

export default vehicleService;